src/osd/OSD.cc

```diff
@@ -2861,13 +2855,13 @@ void OSD::load_pgs()
     dout(10) << "pgid " << pgid << " coll " << coll_t(pgid) << dendl;
     bufferlist bl;
     epoch_t map_epoch = 0;
-    int r = PG::peek_map_epoch(store, pgid, &map_epoch, &bl);
+    ghobject_t pgmeta_oid(pgid.make_pgmeta_oid());
+    int r = PG::peek_map_epoch(store, *it, &map_epoch, &bl, pgmeta_oid);
```

Comment:
We're reusing it later (passing to pg->read_state()). Or I'm not aware of some side effect going on that changes it before reaching pg->read_state()?